packages/form/src/components/Form/index.tsx

```diff
@@ -1,75 +1,146 @@
+  handleSubmit: ReturnType<UseFormHandleSubmit<TFieldValues>>
+}
+
+export const FormSubmitContext = React.createContext<FormSubmitContextValue>(
+  {} as FormSubmitContextValue,
+)
+
+export type FormProps<TFormValues extends FieldValues = FieldValues> = {
+  children?: ((props: FormStateReturn<TFormValues>) => ReactNode) | ReactNode
   errors: FormErrors
-  /**
-   * onRawSubmit is the base onSubmit from final-form
-   */
-  onRawSubmit: ReactFinalFormProps<FormValues, Partial<FormValues>>['onSubmit']
-  initialValues?: Partial<FormValues>
-  validateOnBlur?: ReactFinalFormProps<
-    FormValues,
-    Partial<FormValues>
-  >['validateOnBlur']
-  validate?: ReactFinalFormProps<FormValues, Partial<FormValues>>['validate']
-  /**
-   * The form name attribute
-   */
   name?: string
-  render?: ReactFinalFormProps<FormValues, Partial<FormValues>>['render']
-  mutators?: ReactFinalFormProps<FormValues, Partial<FormValues>>['mutators']
-  keepDirtyOnReinitialize?: boolean
-  className?: string
-}
+  validate?: (
+    values: TFormValues,
+  ) => Partial<Record<FieldPath<TFormValues>, string>>
+  onRawSubmit: (
+    data: TFormValues,
+    formState: {
+      reset: UseFormReturn<TFormValues>['reset']
+      resetFieldState: UseFormReturn<TFormValues>['resetField']
+      restart: () => void
+      change: UseFormReturn<TFormValues>['setValue']
+    },
+  ) => Promise<OnRawSubmitReturn> | OnRawSubmitReturn
+} & XOR<
+  [
+    { initialValues?: DefaultValues<TFormValues> },
+    { methods: UseFormReturn<TFormValues> },
+  ]
+>
 
-export const Form = <FormValues,>({
+export const Form = <TFormValues extends FieldValues>({
   children,
-  onRawSubmit,
-  errors,
+  methods: methodsProp,
   initialValues,
-  validateOnBlur,
-  validate,
+  errors,
+  onRawSubmit,
   name,
-  render,
-  mutators,
-  keepDirtyOnReinitialize,
-  className,
-}: FormProps<FormValues>): JSX.Element => (
-  <ReactFinalForm
-    initialValues={initialValues}
-    validateOnBlur={validateOnBlur}
-    validate={validate}
-    mutators={{
-      ...arrayMutators,
-      ...mutators,
-    }}
-    onSubmit={onRawSubmit}
-    render={
-      render ??
-      (renderProps => (
-        <ErrorProvider errors={errors}>
-          <form
-            noValidate
-            name={name}
-            onSubmit={renderProps.handleSubmit}
-            className={className}
-          >
-            {typeof children === 'function' ? children(renderProps) : children}
+}: FormProps<TFormValues>) => {
+  const methodsHook = useForm({
+    defaultValues: initialValues,
+    mode: 'onChange',
+  })
+
+  const methods = !methodsProp ? methodsHook : methodsProp
+
+  const handleSubmit = methods.handleSubmit(async values => {
+    const result = await onRawSubmit(values, {
+      reset: methods.reset,
+      resetFieldState: methods.resetField,
+      restart: () => methods.reset(initialValues),
+      change: methods.setValue,
+    })
```

Comment:
Do we use somewhere the second parameter ?

Reply:
May be it can be useful in the future but if it is useless actually remove it we will add it later when needed

Reply:
-> Need to be documented as deprecated :)

Reply:
Need to be removed and refactor our forms that use it